workload/scripts/workload-install.sh

```diff
@@ -41,12 +43,28 @@ while [ $# -ne 0 ]; do
     shift
 done
 
+function read_dotnet_link() {
+    cd -P "$(dirname "$1")"
+    dotnet_file="$PWD/$(basename "$1")"
+    while [[ -h "$dotnet_file" ]]; do
+        cd -P "$(dirname "$dotnet_file")"
+        dotnet_file="$(readlink "$dotnet_file")"
+        cd -P "$(dirname "$dotnet_file")"
+        dotnet_file="$PWD/$(basename "$dotnet_file")"
+    done
+    echo $PWD
+}
+
 # Check dotnet install directory.
 if [[ "$DOTNET_INSTALL_DIR" == "<auto>" ]]; then
     if [[ -n "$DOTNET_ROOT" && -d "$DOTNET_ROOT" ]]; then
         DOTNET_INSTALL_DIR=$DOTNET_ROOT
+    elif [[ -d "$DOTNET_DEFAULT_PATH_LINUX" ]]; then
+        DOTNET_INSTALL_DIR=$DOTNET_DEFAULT_PATH_LINUX
+    elif [[ -d "$DOTNET_DEFAULT_PATH_MACOS" ]]; then
+        DOTNET_INSTALL_DIR=$DOTNET_DEFAULT_PATH_MACOS
     else
-        DOTNET_INSTALL_DIR="/usr/share/dotnet"
+        DOTNET_INSTALL_DIR=$(read_dotnet_link $(which dotnet))
```

Comment:
if `which dotnet` can't find `dotnet' in PATH, it will return empty string and `read_dotnet_link` function will return just `PWD`. 
you should check the exit code of `which` command or input of `read_dotnet_link` function.

Reply:
exactly...yes! Thanks for pointing out a missing detail. I just got it updated!